recipe/meta.yaml

```
@@ -4,20 +4,27 @@
 
 {% set cuda_major = environ.get("cuda_compiler_version", "0.0").split(".")[0]|int %}
 
+{% set cuda_present = environ.get("cuda_compiler_version", "None") %}
+{% set kokkos_proc_type = "cpu" if cuda_compiler_version == "None" else "gpu" %}  # [linux64]
+{% set kokkos_proc_type = "cpu" %}                                                # [not linux64]
+
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/kokkos/kokkos/archive/refs/tags/{{ version }}.tar.gz
   sha256: 53cf30d3b44dade51d48efefdaee7a6cf109a091b702a443a2eda63992e5fe0d
-  patches:
-    - msvc.patch  # [win]
+#  patches:
+#    - msvc.patch  # [win]
 
 build:
   # Upstream does not build shared libs for Windows correctly
+{% if cuda_present != "None" %}
   skip: true  # [win]
-  number: 0
+{% endif %}
+  number: 1
+  string: "{{ kokkos_proc_type }}"
```

```suggestion
  # FIXME: Add a jinja variable which holds the build number 
  string: cuda{{ cuda_major }}h{{ PKG_HASH }}_{{ build }}  # [cuda_compiler_version != "None"]
  string: h{{ PKG_HASH }}_{{ build }}  # [cuda_compiler_version == "None"]
```
`kokkos_proc_type` is not unique enough to give future builds a unique build string.

---

perfect,